src/main/scala/org/apache/spark/sql/delta/sources/DeltaSource.scala

```diff
@@ -153,11 +153,7 @@ case class DeltaSource(
   }
 
   private def iteratorLast[T](iter: Iterator[T]): Option[T] = {
-    var last: Option[T] = None
-    while (iter.hasNext) {
-      last = Some(iter.next())
-    }
-    last
+    iter.foldLeft(Option.empty[T])((_, a) => Some(a))
```

Comment:
I would expect the previous implementation to be faster

Reply:
Ok, I doubt. But
```
private def iteratorLast[T](iter: Iterator[T]): Option[T] = {
  @annotation.tailrec
  def loop(acc: Option[T]): Option[T] =
    iter.nextOption() match {
      case None => acc
      case opt => loop(acc = apt)
    }
  loop(acc = None)
}
```
alternatively this can be used.

Reply:
You don't believe me huh? Here's a microbenchmark.
<img width="644" alt="Screen Shot 2020-01-14 at 5 19 17 PM" src="https://user-images.githubusercontent.com/5243515/72396590-07db0080-36f2-11ea-81ae-44e0cc5e9146.png">

I don't like the tailrecursion one either. The current code is readable and efficient. There's no reason to change it.

Reply:
@brkyvz Ah.. Thanks for the guidance. Appreciated. I will revert this back.